components/atoms/Checkbox/checkbox.tsx

```diff
@@ -9,31 +9,34 @@ interface CheckboxProps extends React.ComponentPropsWithoutRef<typeof CheckboxPr
 }
 
 const Checkbox = React.forwardRef<React.ElementRef<typeof CheckboxPrimitive.Root>, CheckboxProps>(
-  ({ className, label, id = "checkbox", ...props }, ref) => (
-    <div className="flex items-center">
-      <CheckboxPrimitive.Root
-        ref={ref}
-        className={clsx(
-          "peer h-4 w-4 shrink-0 rounded-[4px] cursor-pointer bg-white border border-light-slate-8 hover:border-orange-500 focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:border-orange-500 data-[state=checked]:bg-orange-500",
-          className
-        )}
-        {...props}
-        id={id}
-      >
-        <CheckboxPrimitive.Indicator className={clsx("flex items-center justify-center text-white")}>
-          <FiCheck className="w-full h-full" />
-        </CheckboxPrimitive.Indicator>
-      </CheckboxPrimitive.Root>
-      {label && (
-        <label
-          htmlFor={id}
-          className="ml-3 text-sm font-medium leading-none cursor-pointer text-light-slate-12 peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
+  ({ className, label, id, ...props }, ref) => {
+    id = id ? id : label?.replaceAll(" ", "_").toLowerCase();
```

Comment:
This line is the line that I changed here, plus the ID call in the function, yet git did not detect it only (not sure why)

Reply:
You should turn this to a function that returns the conditional statements here

```suggestion
    const getId = ()=> id ? id : label?.replaceAll(" ", "_").toLowerCase();
   
```

Then in the component attributes, set it to `id={getId()}`

Reply:
I am going to implement this, but was wondering why we would take this approach?

Reply:
The variable name needs updating to avoid conflicts with the reserved `id` keyword from the props. 

We could have easily said `const newId = id ? id : label?.replaceAll(" ", "_").toLowerCase();` but it makes more sense with a function since we are performing some checks to determine the used `id`.